internetarchive/utils.py

```diff
@@ -196,7 +198,7 @@ def _get_tag_text(tag_name, xml_obj):
         return str(xml_str)
 
 
-def get_file_size(file_obj):
+def get_file_size(file_obj) -> int | None:
```

Comment:
Maybe Optional[int] should be preferred as the return type? I'm not quite sure about which would be more readable. I guess `int | None` would be more readable to people who had no contact with the type annotations before while `Optional[int]` would be clearer to people who are more used to the type annotations.

Reply:
I do think that `int | None` is more explicit and readable, especially now that `typing.Required` and `typing.NotRequired` is also a thing, albeit in other contexts ([PEP 655](https://peps.python.org/pep-0655/)).

This is [PEP 604](https://peps.python.org/pep-0604/), by the way, and I'm not entirely sure how it behaves on Python 3.7 to 3.9 (though it clearly isn't an error with the `__future__` import).

Reply:
Well, `int | None` is certainly more explicit or obvious.

Although I think that `Optional[int]` communicates the intent better and also fits better with the practices in type theory and type systems of other languages like ML, Haskell or C++.
So I'm undecided on this.

`None` is essentially often just a value used as input or output if something is optional and it does not really matter that it is `None`, but rather that it is optional.

Reply:
`pyupgrade --py310-plus` will remove the `annotations` import  and then drop both `typing.Optional` and `typing.Union` so let’s go that way too.

Reply:
Interesting. I did not know that Optional would be deprecated or at least discouraged.
But lets definitely go with `|` then.

Reply:
https://github.com/asottile/pyupgrade#pep-604-typing-rewrites